newrelic-agent/src/test/java/com/newrelic/agent/logging/FileAppenderFactoryTest.java

```diff
@@ -0,0 +1,92 @@
+        Assert.assertTrue((policy.getTriggeringPolicies()[0] instanceof CronTriggeringPolicy) ||
+                (policy.getTriggeringPolicies()[0] instanceof SizeBasedTriggeringPolicy));
+        Assert.assertTrue((policy.getTriggeringPolicies()[1] instanceof CronTriggeringPolicy) ||
+                (policy.getTriggeringPolicies()[1] instanceof SizeBasedTriggeringPolicy));
+    }
+
+    @Test
+    public void build_withLogDailyTrueAndByteLimitEqualZero_createsDailyRollingFileAppender() {
+        FileAppenderFactory factory = new FileAppenderFactory(1, 0, FILENAME + "1", true);
+        AbstractOutputStreamAppender<? extends FileManager> fileAppender = factory.build();
+        RollingFileAppender rollingFileAppender = (RollingFileAppender) fileAppender;
+
+        Assert.assertEquals(RollingFileAppender.class, fileAppender.getClass());
+        Assert.assertEquals(CompositeTriggeringPolicy.class, rollingFileAppender.getTriggeringPolicy().getClass());
+        Assert.assertEquals(FILENAME + "1", rollingFileAppender.getFileName());
+
+        CompositeTriggeringPolicy policy = rollingFileAppender.getTriggeringPolicy();
+        Assert.assertTrue((policy.getTriggeringPolicies()[0] instanceof CronTriggeringPolicy) ||
+                (policy.getTriggeringPolicies()[0] instanceof NoOpTriggeringPolicy));
+        Assert.assertTrue((policy.getTriggeringPolicies()[1] instanceof CronTriggeringPolicy) ||
+                (policy.getTriggeringPolicies()[1] instanceof NoOpTriggeringPolicy));
+    }
+
+    @Test
+    public void build_withLogDailyFalseAndByteLimitGreaterThanZero_createsSizeBasedRollingFileAppender() {
+        FileAppenderFactory factory = new FileAppenderFactory(1, BYTE_LIMIT, FILENAME + "2", false);
+        AbstractOutputStreamAppender<? extends FileManager> fileAppender = factory.build();
+        RollingFileAppender rollingFileAppender = (RollingFileAppender) fileAppender;
+
+        Assert.assertEquals(RollingFileAppender.class, fileAppender.getClass());
+        Assert.assertEquals(FILENAME + "2", rollingFileAppender.getFileName());
+
+        SizeBasedTriggeringPolicy policy = rollingFileAppender.getTriggeringPolicy();
+        Assert.assertEquals(SizeBasedTriggeringPolicy.class, policy.getClass());
+        Assert.assertEquals(BYTE_LIMIT, policy.getMaxFileSize());
+    }
+
+    @Test
+    public void build_withLogDailyFalseAndByteLimitEqualZero_createsSizeBasedRollingFileAppender() {
+        FileAppenderFactory factory = new FileAppenderFactory(1, 0, FILENAME + "3", false);
+        AbstractOutputStreamAppender<? extends FileManager> fileAppender = factory.build();
+        FileAppender rollingFileAppender = (FileAppender) fileAppender;
+
+        Assert.assertEquals(FileAppender.class, fileAppender.getClass());
+        Assert.assertEquals(FILENAME + "3", rollingFileAppender.getFileName());
+
+        try {
+            System.out.println(new File(FILENAME + "0").getCanonicalPath());
```

Comment:
Should these be asserts?
Or is this just debug code?

If not debug code and these throw an exception, it should fail the test.

Reply:
It was debug I forgot to remove...